src/transformers/models/clvp/modeling_clvp.py

```diff
@@ -0,0 +1,1338 @@
+        >>> from transformers import CLVPTokenizer, CLVPModel
+
+        >>> model = CLVPModel.from_pretrained("susnato/clvp_dev")
+        >>> tokenizer = CLVPTokenizer.from_pretrained("susnato/clvp_dev")
+
+        >>> inputs = tokenizer(["a photo of a cat", "a photo of a dog"], padding=True, return_tensors="pt")
+        >>> text_features = model.get_text_features(**inputs)
+        ```"""
+
+        # Use CLVP model's config for some fields (if specified) instead of those of speech & text components.
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        text_outputs = self.text_model(
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+            use_causal_attention_mask=use_causal_attention_mask,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        pooled_output = text_outputs[1]
+        text_features = self.text_projection(pooled_output)
+
+        return text_features
+
+    @add_start_docstrings_to_model_forward(CLVP_SPEECH_INPUTS_DOCSTRING)
+    def get_speech_features(
+        self,
+        speech_ids: Optional[torch.Tensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        use_causal_attention_mask: Optional[bool] = False,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> torch.FloatTensor:
+        r"""
+        Returns:
+            speech_features (`torch.FloatTensor` of shape `(batch_size, output_dim)`: The speech embeddings obtained by
+            applying the projection layer to the pooled output of [`CLVPSpeechModel`].
+
+        Examples:
+
+        ```python
+        >>> from transformers import CLVPModel
+
+        >>> model = CLVPModel.from_pretrained("susnato/clvp_dev")
+
+        >>> # TODO : after FeatureExtractor is implemented we need to change it to something lik FE.__call__(...)
+        >>> inputs = {"speech_ids": torch.tensor([[56, 8, 48, 7, 11, 23]]).long()}
+
+        >>> speech_features = model.get_speech_features(**inputs)
+        ```"""
+
+        # Use CLVP model's config for some fields (if specified) instead of those of speech & text components.
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
```

Comment:
IMO, and as discussed in my review, you should add the model that process MEL spectrograms to those `speech_ids` in your model.

It could be initalized in the `CLVPModel.__init__` and called like:

```python

speech_ids = self.speech_prenet(
    input_mel_spectrograms,
    ...
)

speech_outputs = self.speech_model(
            input_ids=speech_ids,
            attention_mask=attention_mask,
            use_causal_attention_mask=use_causal_attention_mask,
            output_attentions=output_attentions,
            output_hidden_states=output_hidden_states,
            return_dict=return_dict,
        )
```

Reply:
This `speech_prenet` could be simple an additional `CLVPTransformer` (but I doubt it since it takes MEL spectrograms as input), you should verify it